### ISSUE TITLE
Opening a new tab in an existing instance from shell

### ISSUE THREAD

**Issue by corvidism** [geany/geany#1512]
Hi,

is it possible to start creating a new file in an existing instance from shell? I'd like to use a keyboard shortcut to start a new file, but `geany` with no arguments opens a new instance (not useful).

If it can't be done, how likely are you to implement it? Ideally, this would be a commandline parameter (something like the opposite of `-i`).

**Comment by Bfgeshka** [comment]
It should be opened in an opened window if it exists. I always have only one instance.

**Comment by corvidism** [comment]
Hm. When I call `geany` from shell (either from a new one or using `geany &` repeatedly) it spawns a new instance every time.

When I call `geany <filename>` it opens the file in the first instance that was opened. (Thinking of which, that also isn't too good - I'd expect it to appear in the last one).

**Comment by elextr** [comment]
No geany cannot open a new file from the command line, if a filename is
given geany attempts to open it and complains if it doesn't exist since its
most likely a typo.  A command line option to specify this is a new file
would be needed, but nobody has written it.

Running Geany on the command line without a filename creates a new instance
as you have observed, this is intended behaviour documented in the manual.

Running Geany from the command line with at least one filename more than
once defaults to opening all files in the first instance as you observed,
this is also intended and documented behaviour.  If you want a new instance
use the -i option.

I always warn about running more than one instance of Geany using the same
configuration, there is no sharing of changes and the last one to write its
version of the configuration "wins" at shutdown.

Cheers
Lex

**Comment by codebrainz** [comment]
> No geany cannot open a new file from the command line, if a filename is given geany attempts to open it and complains if it doesn't exist since its most likely a typo.

Here it behaves just like nano or vim and opens a new unsaved file with the given filename.

**Comment by elextr** [comment]
@codebrainz weird

```
lex@fred7 ~ $ ./geany -c  /tmp/foo
bash: ./geany: No such file or directory
lex@fred7 ~ $
```

**Comment by elextr** [comment]
Ahh, couldn't find geany, ok, now it works :)

**Comment by codebrainz** [comment]
Yeah, that's the usual behaviour, it would be annoying if created the file, for example if you made or a typo or decided you didn't want to save the file, you'd have to go into terminal/file manager and manually delete it.

I think this issue could probably be closed as you can do what the title says already.

**Comment by vfaronov** [comment]
You can use the `--socket-file` option to open a file in a specific instance of Geany. For example, if you execute these two commands:

```
geany --socket-file=/tmp/geany1 /tmp/foo
geany --socket-file=/tmp/geany1 /tmp/bar
```

you get one instance of Geany with two open documents, `/tmp/foo` and `/tmp/bar`.

Here’s a script I use (on Linux) to have one instance of Geany per workspace: https://gist.github.com/vfaronov/6843c6bc817521bfab7e58fff139c525

**Comment by corvidism** [comment]
Thanks for the replies, but none of the suggestions are helpful with what I'm trying to do, which is to **open a new tab** (empty "untitled" file, or a new buffer as @elextr said) **in an existing instance** (any existing instance, I just don't want to spawn a new one).

> technically it opens a new buffer, it doesn't actually create the file until you save the buffer

And thanks for that, because I tried calling `geany ""` and it started editing `/home/<username>` which... ouch. XD

> I think this issue could probably be closed as you can do what the title says already.

...no, I can't? Unless you choose to interpret it as "open existing file in a new tab" which, okay, but not what I meant.

Oh, well. Is there any official way of submitting a feature request for this?

**Comment by elextr** [comment]
> Oh, well. Is there any official way of submitting a feature request for this?

You have :smile:

**Comment by codebrainz** [comment]
I don't get it, you can open a new empty file from the command line, and @vfaronov showed how to open that new file in any existing instance instead of just the first one. What more is needed?

**Comment by codebrainz** [comment]
@corvidism is it that you don't want to choose a name for the new file up-front but rather later using the Save As dialog?

**Comment by codebrainz** [comment]
If so, [this patch](https://gist.github.com/codebrainz/71d761e2cfd7e7519106ce4f731dd5ee) should work, but it doesn't since Geany ignores request to use existing socket file when no filenames are given, so it always opens a new instance. I have no idea where that code is, but something like this should work with the patch if it was fixed:

```bash
$ geany -u
# or
$ geany --socket-file=/what/ever -u
```

**Comment by corvidism** [comment]
> is it that you don't want to choose a name for the new file up-front but rather later using the Save As dialog?

Yes, that's it.

**Comment by b4n** [comment]
@corvidism @LarsGit223 has implemented this in #1811, but I'm wondering what is the actual use case for *not* choosing the filename? It eludes me why one would like to create a new empty document from outside Geany without anything, be it content or even path.
Once I'm convinced it does have a use case, I'll properly test and merge the mentioned PR (which looks OK).
